scripts/extensions/videoEditor/src/VideoEditorThumbnail.tsx

```diff
@@ -0,0 +1,378 @@
+import * as React from 'react';
+import {IArticle, ISuperdesk} from 'superdesk-api';
+import {IErrorMessage, ICrop, IVideoProject} from './interfaces';
+import {pick} from 'lodash';
+
+interface IProps {
+    video: HTMLVideoElement;
+    article: IArticle;
+    crop: ICrop;
+    rotate: number;
+    superdesk: ISuperdesk;
+    renditions: IArticle['renditions'];
+    onToggleLoading: (isLoading: boolean, loadingText?: string, type?: 'video' | 'thumbnail') => void;
+    onSave: (renditions: IArticle['renditions']) => void;
+    onError: (err: IErrorMessage) => void;
+    getCropRotate: (crop: ICrop) => ICrop;
+}
+
+interface IState {
+    type: 'capture' | 'upload' | '';
+    value: number | File; // capture positon or uploaded File
+    rotateDegree: number; // save current rotate degree when user captures thumbnail
+    scale: number;
+}
+
+const initialState: IState = {type: '', value: 0, rotateDegree: 0, scale: 1};
+
+export class VideoEditorThumbnail extends React.Component<IProps, IState> {
+    private ref: React.RefObject<HTMLCanvasElement>;
+    private maxCanvasSize: { width: number; height: number };
+    private interval: number;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = initialState;
+        this.ref = React.createRef();
+        this.maxCanvasSize = {width: 0, height: 0};
+        this.interval = 0;
+
+        this.handleClick = this.handleClick.bind(this);
+        this.handleUpload = this.handleUpload.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+        this.handleReset = this.handleReset.bind(this);
+        this.getWrapperSize = this.getWrapperSize.bind(this);
+    }
+
+    componentDidMount() {
+        const thumbnail = this.props.renditions?.thumbnail?.href;
+
+        if (thumbnail) {
+            this.setThumbnail(thumbnail);
+        }
+    }
+
+    componentDidUpdate(prevProps: IProps) {
+        // video has changed so captured thumbnail showed on UI is no longer correct
+        if (prevProps.renditions !== this.props.renditions && this.state.type === 'capture') {
```

Comment:
```this.state.type === 'capture'```

what is special with the capture state? would it work to instead check if `state.capture != prevState.capture`?

Reply:
No we are not checking `capture` state. We are checking if video has changed or not.
This is when you capture thumbnail but do not click save immediately, but _after editing video_, in that case, current display thumbnail will be differ from the captured. In case upload thumbnail, it doesn't depend on video.

We don't re-draw thumbnail because the video maybe has not finished loading yet, we could wait but reset it is easier